jaxrs/WFLY-12779_Use-MicroProfile-Config-in-RESTEasy

```diff
@@ -0,0 +1,77 @@
+= Use MicroProfile Config in RESTEasy
+:author:            Ron Sigal
+:email:             rsigal@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, RESTEasy 3.x (targeted to Wildfly) uses servlet context-params to obtain
+configuration information. In RESTEasy 4.x, we have updated configuration to access
+all standard MicroProfile Config sources. Also, we have added three additional ConfigSources:
+
+* servlet init-params [ordinal 60]
+* filter init-params [ordinal 50]
+* servlet context-params [ordinal 40]
+
+In RESTEasy 4.x, all parameters, both RESTEasy parameters and application parameters, are
+available from all ConfigSources to both RESTEasy and application code. In RESTEasy 3.x, we
+propose to add a RESTEasy parameter, "resteasy.config.resteasy.parameters.only", which, when
+set to "true", will limit the three RESTEasy ConfigSources to returning RESTEasy parameters
+only. It will default to "true".
+
+== Issue Metadata
+
+=== Issue
+
+* {issue-base-url}/EAP7-1386 [EAP7-1386]
+* {issue-base-url}/WFLY-12779 [WFLY-12779]
+
+=== Related Issues
+
+* {issue-base-url}/RESTEASY-2406 [RESTEASY-2406]
+
+=== Dev Contacts
+
+* mailto:rsigal@redhat.com[Ron Sigal]
+* mailto:asoldano@redhat.com[Alessio Soldano]
+
+=== QE Contacts
+
+* ??
+
+=== Testing By
+
+[x] Engineering
```

Comment:
This RFE should be tested also by QE, asking for bump of the rank.

Reply:
Updated